### PR TITLE
Schema validation error logging bug

### PR DESCRIPTION
**Pull request by jhowlett-scottlogic**
PR for https://github.com/ScottLogic/openapi-forge/issues/49

- Add test to see if errors object is loopable. 
- Add test to see if errors.instancePath is defined before adding it to error message

### CODE REVIEW

File: src/generate.js [hunk 1]
```diff
@@ -60,12 +60,25 @@ async function isValidSchema(schema) {
     await SwaggerParser.validate(cloneSchema(schema));
   } catch (errors) {
     console.error("Schema validation errors:");
-    errors.forEach((error) => {
-      console.error(`${error.message} at ${error.instancePath}`);
-    });
+    let errorMessage;
+    if(errors.length !== undefined) {
```
Comment (ColinEberhardt) [inline src/generate.js:64]: I'd use [Array.isArray](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray) - the intent is much clearer

File: src/generate.js [hunk 2]
```diff
@@ -60,12 +60,25 @@ async function isValidSchema(schema) {
...
+      errors.forEach((error) => {
+        errorMessage = error.message;
+        if(error.instancePath != undefined) {
+          errorMessage +=  `at ${error.instancePath}`
+        }
+        console.error(errorMessage);
+      });
+    } else {
+      errorMessage = errors.message;
```
Comment (ColinEberhardt) [inline src/generate.js:73]: I'm not keen on the duplication of logic here. How about this ...

~~~
const errorArray = Array.isArray(errors) ? errors : [errors];
~~~

Then just use your logic for iterating an array of errors?

**Review by ColinEberhardt — CHANGES_REQUESTED** [review]
Thanks - a few cosmetic suggestions ...

**Comment by jhowlett-scottlogic** [comment]
@ColinEberhardt I have addressed your comments and also moved logging on error over to the new format

**Comment by ColinEberhardt** [comment]
Much better :-)

**Review by ColinEberhardt — APPROVED** [review]

**Pull request merged**